src/libraries/System.Security.Cryptography/tests/X509Certificates/X500DistinguishedNameBuilderTests.cs

```diff
@@ -0,0 +1,379 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Formats.Asn1;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    public static class X500DistinguishedNameBuilderTests
+    {
+        private const string TestOid = "2.25.77135202736018529853602245419149860647";
+
+        [Fact]
+        public static void AddEmailAddress_Success_EncodedToIA5()
+        {
+            AssertBuilder(
+                "30223120301E06092A864886F70D01090116116B6576696E406578616D706C652E636F6D",
+                builder => builder.AddEmailAddress("kevin@example.com"));
+        }
+
+        [Fact]
+        public static void AddEmailAddress_ExceedsMaximumLength_Fails()
+        {
+            X500DistinguishedNameBuilder builder = new();
+            string bigEmail = new string('a', 244) + "@example.com";
```

Comment:
We could make this test simpler, and avoid math, by just using `new string('a', 256)`, since the implementation doesn't look for `@`.

Reply:
It seemed more sensible to me to test the "real" thing, similar to your sentiment further below:

> I certainly don't want to make this a negative test, but, IIRC, it's supposed to be the full name of the state, not the abbreviation. So "California"

But we can add a test that gibberish is accepted.